### PR TITLE
add gamification integration for kudos

### PR DESCRIPTION
**Pull request by nerminebouj**

### CODE REVIEW

**Review by boubaker — CHANGES_REQUESTED** [review]
Please add Gamification integration listener in a dedicated Class without changing the existing listener that generates the activity.

**Review by AmineChalouati — CHANGES_REQUESTED** [review]
In Integration Gamification Listener. Kindly relate object with the link to the concerned activity  gam.put("object", ""); to broadcast this event.
In the Listener broadcast kindly put the root to the activity .
       listenerService.broadcast("exo.gamification.generic.action", gam, "");

File: exo-kudos-services/src/main/java/org/exoplatform/addon/kudos/listener/IntegrationGamificationListener.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      gam.put("receiverId", kudos.getSenderId());
+      gam.put("object", "");
+      listenerService.broadcast("exo.gamification.generic.action", gam, "");
+  } catch (Exception e) {
+    LOG.error("Cannot broadcast gamification event");
+  }
+    try {
+      Map<String, String> gam = new HashMap<>();
+      gam.put("ruleTitle", "receiveKudos");
+      gam.put("senderId", kudos.getReceiverId());
```
Comment (mkrout) [inline exo-kudos-services/src/main/java/org/exoplatform/addon/kudos/listener/IntegrationGamificationListener.java]: I think you should put the kudos sender here:
gam.put("senderId", kudos.getSenderId());

Ps in gamification the sender is user who do the action (here the one who send the kudo) and the receiver id the user who will get the points (here the one who receive the kudo)

And please be sure thet the senderId and receiverId in the gamification object should be the userName not the social identity ID

**Review by mkrout — CHANGES_REQUESTED** [review]

File: exo-kudos-services/src/main/java/org/exoplatform/addon/kudos/listener/IntegrationGamificationListener.java (hunk 1 above)
Comment (boubaker) [reply exo-kudos-services/src/main/java/org/exoplatform/addon/kudos/listener/IntegrationGamificationListener.java]: done

**Review by boubaker — APPROVED** [review]

**Pull request merged**